sdk/communication/communication-common/src/identifierModelSerializer.ts

```diff
@@ -150,6 +154,22 @@ export const serializeCommunicationIdentifier = (
   }
 };
 
+const getKind = (serializedIdentifier: SerializedCommunicationIdentifier): string => {
+  if (serializedIdentifier.communicationUser) {
+    return "communicationUser";
+  }
+
+  if (serializedIdentifier.phoneNumber) {
+    return "phoneNumber";
+  }
+
+  if (serializedIdentifier.microsoftTeamsUser) {
+    return "microsoftTeamsUser";
```

Comment:
@petrsvihlik  @AikoBB I also didn't like this duplication and created [exported constants](https://github.com/Azure/azure-sdk-for-js/pull/23504/commits/e3b772fb5be92bf9e94a4165e57f7357eaf1ed08) for it. It caused pipelines to fail. Unfortunately, the build is no longer available, and I don't remember the exact error. Such change will also make interfaces less understandable because we'll need to replace strings with the identifier type (like "microsoftTeamsUser") with a string type or `typeof`, which will make them less explicit. I concluded that such repetition is part of the JS architecture. If it is not so, I could make further investigation.

Reply:
this would require bigger changes, also an approval from architects, and it wouldn't be idiomatic to how we use these constants right now in TS. we agreed to skip this for now.